File: process-mgm/chunks/04-getpid.go
package main

import (
	"fmt"
	"os"
	"syscall"
)

func main() {
	// просто обертка для syscall.Getpid():
	// func Getpid() int { return syscall.Getpid() }
	currentPID := os.Getpid()
	fmt.Println("My PID: ", currentPID)

	// просто обертка для syscall.Getppid():
	// func Getppid() int { return syscall.Getppid() }
	parentPID := os.Getppid()
	fmt.Println("Parent's PID: ", parentPID)

	currentPID = syscall.Getpid()
	parentPID = syscall.Getppid()
	fmt.Println("PID: ", currentPID)
	fmt.Println("PPID: ", parentPID)
}

File: process-mgm/chunks/04-getpid.c
#include <sys/types.h> // можно и не импортировать, если getpid() закомменчен
#include <unistd.h>
#include <stdio.h>

pid_t getpid(void); // можно и закомментировать или удалить - все равно работает

int main() {
    printf("My pid = %jd\n", (__intmax_t) getpid());
    printf("Parent's pid = %jd\n", (__intmax_t) getppid()); 

    return 0;
}
